includes/Modules/Analytics.php

```diff
@@ -937,115 +937,107 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 						'properties' => array(),
 						'profiles'   => array(),
 					);
 
-					$found_account_id = false;
-					$matched_property = false;
 					$existing_tag     = $this->_existing_tag_account;
+					$found_account_id = ! empty( $existing_tag['accountId'] ) ? $existing_tag['accountId'] : false;
+					unset( $this->_existing_tag_account );
 
-					$this->_existing_tag_account = null; // Set back to null.
-
-					if ( empty( $existing_tag ) ) {
+					if ( ! $found_account_id ) {
+						// Get the account ID from the saved settings - returns WP_Error if not set.
 						$account_id = $this->get_data( 'account-id' );
-						if ( ! is_wp_error( $account_id ) ) {
-							foreach ( $response['accounts'] as $account ) {
-								if ( $account->getId() === $account_id ) {
-									$found_account_id = $account->getId();
-									break;
-								}
-							}
+						// If the saved account ID is in the list of accounts the user has access to, it's a match.
+						if ( in_array( $account_id, $account_ids, true ) ) {
+							$found_account_id = $account_id;
 						} else {
-							$current_url = untrailingslashit( $this->context->get_reference_site_url() );
-							$urls        = $this->permute_site_url( $current_url );
-							foreach ( $response['accounts'] as $account ) {
-								$properties = $this->get_data( 'properties-profiles', array( 'accountId' => $account->getId() ) );
-								if ( is_wp_error( $properties ) ) {
-									continue;
-								}
-								$url_matches = array_filter(
-									$properties['properties'],
-									function( $property ) use ( $urls ) {
-										return in_array( untrailingslashit( $property->getWebsiteUrl() ), $urls, true );
-									}
-								);
-								if ( ! empty( $url_matches ) ) {
-									$found_account_id = $account->getId();
-									$matched_property = $url_matches;
-									break;
+							foreach ( $accounts as $account ) {
+								$properties_profiles = $this->get_data( 'properties-profiles', array( 'accountId' => $account->getId() ) );
+
+								if ( ! is_wp_error( $properties_profiles ) && isset( $properties_profiles['matchedProperty'] ) ) {
+									return array_merge( $response, $properties_profiles );
 								}
 							}
 						}
-					} else {
-						$found_account_id = $existing_tag['accountId'];
-					}
-
-					if ( empty( $found_account_id ) ) {
-						return $response;
 					}
 
-					$properties = $this->get_data( 'properties-profiles', array( 'accountId' => $found_account_id ) );
-					if ( is_wp_error( $properties ) ) {
+					if ( ! $found_account_id ) {
 						return $response;
 					}
 
-					$result = array_merge( $response, $properties );
-
-					// Get matched property from exiting tag property id.
-					if ( ! empty( $existing_tag ) ) {
-						$matched_property = array_filter(
-							$properties['properties'],
-							function( $property ) use ( $existing_tag ) {
-								return $property->getId() === $existing_tag['propertyId'];
-							}
-						);
-					}
```

Comment:
Removing this is problematic since you didn't cover for it in any way in `properties-profiles`. We need to continue to ensure that, if an existing tag is present, the property from that tag is the `matchedProperty`.